### PR TITLE
Sroc update EA charge bands

### PR DESCRIPTION
**Pull request by jaswinder2303**
Product Backlog Item 68472:SROC: Update EA charge bands * BUSINESS Q

### CODE REVIEW

File: src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandCalculatorChooser.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        private readonly IEnvironmentAgencyProducerChargeBandCalculator environmentAgencyProducerChargeBandCalculator;
+        private readonly IProducerChargeBandCalculator producerChargeBandCalculator;
+
+        public ProducerChargeBandCalculatorChooser(IEnvironmentAgencyProducerChargeBandCalculator environmentAgencyProducerChargeBandCalculator, IProducerChargeBandCalculator producerChargeBandCalculator)
+        {
+            this.environmentAgencyProducerChargeBandCalculator = environmentAgencyProducerChargeBandCalculator;
+            this.producerChargeBandCalculator = producerChargeBandCalculator;
+        }
+
+        public IProducerChargeBandCalculator GetCalculator(schemeType scheme, producerType producer, int complianceYear)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandCalculatorChooser.cs]: Can we add some tests for the chooser?

File: src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EA.Weee.Xml.MemberRegistration
+{
+    class ProducerChargeBandStrategy
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandStrategy.cs]: Can we delete this now?

File: src/EA.Weee.Xml/MemberRegistration/Extensions.cs [hunk 3]
```diff
@@ -24,6 +24,29 @@ public static string GetProducerName(this producerType producer)
             return producer.tradingName;
         }
 
+        public static countryType GetProducerCountry(this producerType producer)
+        {
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Xml/MemberRegistration/Extensions.cs]: Create some unit tests to test these...

File: src/EA.Weee.RequestHandlers/Scheme/MemberRegistration/ProducerChargeCalculator.cs [hunk 4]
```diff
@@ -7,34 +7,34 @@
...
+            this.producerChargeBandCalculatorChooser = producerChargeBandCalculatorChooser;
         }
 
-        public ProducerCharge CalculateCharge(string schemeApprovalNumber, producerType producer, int complianceYear)
+        public ProducerCharge CalculateCharge(schemeType scheme, producerType producer, int complianceYear)
         {
-            ProducerCharge producerCharge = GetProducerCharge(producer);
+            ProducerCharge producerCharge = GetProducerCharge(scheme, producer, complianceYear);
 
             if (producer.status == statusType.A)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Scheme/MemberRegistration/ProducerChargeCalculator.cs]: Just looking at the acceptance criteria 
7.Remove / comment out the code that deals with changes to charge bands (we'll update this in a separate story)
We should comment out this checking for A status... I think there will be a new rule coming in

File: src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandStrategy.cs (hunk 2 above)
Comment (jaswinder2303) [reply src/EA.Weee.Xml/MemberRegistration/ProducerChargeBandStrategy.cs]: yes will do it now

File: src/EA.Weee.RequestHandlers/Scheme/MemberRegistration/ProducerChargeCalculator.cs (hunk 4 above)
Comment (jaswinder2303) [reply src/EA.Weee.RequestHandlers/Scheme/MemberRegistration/ProducerChargeCalculator.cs]: commented out

**Pull request merged**